### PR TITLE
LIVY-264. Fixed PySpark %table magic returning wrong header.

### PR DESCRIPTION
**Pull request by purechoc**
%table returns wrong header if input is a list of rows.
```
from pyspark.sql.types import Row
x = [Row(age=1, name=u'a'), Row(age=2, name=u'b'), Row(age=3, name=u'c')]
%table x
```
```
{
  ...
        "data": {
          "application/vnd.livy.table.v1+json": {
            "headers": [
              {
                "type": "INT_TYPE",
                "name": "0"
              },
              {
                "type": "STRING_TYPE",
                "name": "1"
              }
  ...
```
Notice that the header names are wrong.
Workaround the issue by converting row to dict.

### CODE REVIEW

**Comment by tc0312** [comment]
Mind if you add test?

**Comment by purechoc** [comment]
@tc0312 
there is any way to import pyspark library in unitest?

**Comment by tc0312** [comment]
You can do it in integration test:
https://github.com/cloudera/livy/blob/master/integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala#L66

**Comment by purechoc** [comment]
ok. i'll try make uniitest

**Comment by purechoc** [comment]
add integration test complete.

File: integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala [hunk 1]
```diff
@@ -69,7 +69,10 @@ class InteractiveIT extends BaseIntegrationTestSuite {
       s.run("sqlContext").verifyResult(startsWith("<pyspark.sql.context.HiveContext"))
       s.run("sc.parallelize(range(100)).map(lambda x: x * 2).reduce(lambda x, y: x + y)")
         .verifyResult("9900")
-
+      s.run("from pyspark.sql.types import Row").verifyResult(".*")
```
Comment (tc0312) [inline integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala]: verifyResult(".*") is a bit too board. Mind if you restrict it a bit?

File: repl/src/main/resources/fake_shell.py [hunk 2]
```diff
@@ -403,6 +406,9 @@ def magic_table(name):
         cols = []
         data.append(cols)
 
+        if os.environ.get("LIVY_TEST") != "true" and isinstance(row, pyspark.sql.types.Row):
```
Comment (tc0312) [inline repl/src/main/resources/fake_shell.py]: Mind if you explain what's `os.environ.get("LIVY_TEST") != "true"` for?

Comment (purechoc) [inline repl/src/main/resources/fake_shell.py]: when os.environ.get("LIVY_TEST") == "true"
fake_shell do not import spark library (can not find pyspark.sql.types.Row)

File: integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala (hunk 1 above)
Comment (purechoc) [inline integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala]: got it. i'll fix it

File: repl/src/main/resources/fake_shell.py (hunk 2 above)
Comment (tc0312) [inline repl/src/main/resources/fake_shell.py]: I think we can do the type checking without importing the class by doing string comparsion on row.__class__.__name__.

An alternative is, we can just check does asDict() present in the row object. If it does, then call it.

File: integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala [hunk 3]
```diff
@@ -69,9 +69,9 @@ class InteractiveIT extends BaseIntegrationTestSuite {
       s.run("sqlContext").verifyResult(startsWith("<pyspark.sql.context.HiveContext"))
       s.run("sc.parallelize(range(100)).map(lambda x: x * 2).reduce(lambda x, y: x + y)")
         .verifyResult("9900")
-      s.run("from pyspark.sql.types import Row").verifyResult(".*")
+      s.run("from pyspark.sql.types import Row").verifyResult("")
       s.run("x = [Row(age=1, name=u'a'), Row(age=2, name=u'b'), Row(age=3, name=u'c')]")
```
Comment (tc0312) [inline integration-test/src/test/scala/com/cloudera/livy/test/InteractiveIT.scala]: Sorry mind if you explain what you are fixing?
I ran these code without your fix but %table magic is still working.

File: repl/src/main/resources/fake_shell.py (hunk 2 above)
Comment (purechoc) [inline repl/src/main/resources/fake_shell.py]: 'string comparison' looks better. I will apply it

**Comment by tc0312** [comment]
Merging once test passes.

**Pull request merged**